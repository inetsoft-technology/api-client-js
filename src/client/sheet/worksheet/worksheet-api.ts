/*
 * Copyright 2019 InetSoft Technology
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import { Sheet } from "../sheet-api";

/**
 * WorksheetParameter describes a parameter used when opening a worksheet.
 */
export interface WorksheetParameter {
   /**
    * The name of the parameter.
    */
   name: string;

   /**
    * The data type of the parameter. The allowable values are 'boolean', 'float', 'double',
    * 'character', 'byte', 'short', 'integer', 'long', 'date', 'time', and 'datetime'.
    */
   type: string;
}

/**
 * TableDataRow contains the data of a single worksheet row.
 */
export interface TableDataRow {
   /**
    * The data values in the row. The actual type of these values may be a string, number or
    * boolean.
    */
   values: any[];
}

/**
 * TableData contains rows of worksheet data.
 */
export interface TableData {
   /**
    * The number of rows in the table.
    */
   rowCount: number;

   /**
    * The number of columns in the table.
    */
   colCount: number;

   /**
    * The number of header rows in the table.
    */
   headerRowCount: number;

   /**
    * The number of header columns in the table.
    */
   headerColCount: number;

   /**
    * The number of footer rows in the table.
    */
   trailerRowCount: number;

   /**
    * The number of summary columns in the table.
    */
   trailerColCount: number;

   /**
    * Unique identifiers for each column.
    */
   colIdentifiers: string[];

   /**
    * The data types for each column. The possible values are 'boolean', 'float', 'double',
    * 'character', 'byte', 'short', 'integer', 'long', 'date', 'time', and 'datetime'.
    */
   colTypes: string[];

   /**
    * The rows of data.
    */
   rows: TableDataRow[];
}

/**
 * WorksheetApi provides operations to access worksheets.
 */
export interface WorksheetApi {
   /**
    * Gets the list of available worksheets.
    *
    * @return the list of worksheets.
    */
   getWorksheets(): Promise<Sheet[]>;

   /**
    * Gets the parameters for a worksheet.
    *
    * @param id the asset identifier of the worksheet.
    *
    * @return the worksheet parameters.
    */
   getWorksheetParameters(id: string): Promise<WorksheetParameter[]>;

   /**
    * Gets the meta data for a worksheet.
    *
    * @param id the asset identifier of the worksheet.
    *
    * @return the worksheet metadata.
    */
   getWorksheetMetadata(id: string): Promise<TableData>;

   /**
    * Gets the list of opened worksheets.
    *
    * @return the list of opened worksheets.
    */
   getOpenedWorksheets(): Promise<Sheet[]>;

   /**
    * Opens a worksheet.
    *
    * @param asset      the asset identifier of the worksheet.
    * @param parameters the parameters to use when opening the sheet.
    *
    * @return the worksheet was opened successfully.
    */
   openWorksheet(asset: string, parameters: {[name: string]: any}): Promise<Sheet>;

   /**
    * Gets information about an opened worksheet.
    *
    * @param id the runtime identifier of the opened worksheet.
    *
    * @return the opened worksheet information.
    */
   getOpenedWorksheet(id: string): Promise<Sheet>;

   /**
    * Closes an opened worksheet.
    *
    * @param id the runtime identifier of the opened worksheet.
    */
   closeWorksheet(id: string): Promise<void>;

   /**
    * Gets a block of rows from an opened worksheet.
    *
    * @param id   the runtime identifier of the opened worksheet.
    * @param from the starting, zero-based row number for the returned data. If not specified, 0
    *             will be used.
    * @param rows the number of rows to return. Fewer rows may be returned if the end of the table
    *             is reached.
    *
    * @return the worksheet data rows.
    */
   getWorksheetData(id: string, from: number, rows: number): Promise<TableData>;
}
