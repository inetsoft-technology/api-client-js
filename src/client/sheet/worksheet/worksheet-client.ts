/*
 * Copyright 2019 InetSoft Technology
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import { fromByteArray } from "base64-js";
import { AbstractClient } from "../../abstract-client";
import { HttpAdapter } from "../../http-adapter";
import { Sheet } from "../sheet-api";
import { TableData, WorksheetApi, WorksheetParameter } from "./worksheet-api";

/**
 * @hidden
 */
export class WorksheetClient extends AbstractClient implements WorksheetApi {

   private static encodeId(id: string): string {
      const utf8String = unescape(encodeURIComponent(id));
      const utf8Array = new Uint8Array(utf8String.length);

      for (let i = 0; i < utf8String.length; i++) {
         utf8Array[i] = utf8String.charCodeAt(i);
      }

      return fromByteArray(utf8Array);
   }

   constructor(adapter: HttpAdapter<any>, token: () => Promise<string>, baseUrl: string) {
      super(adapter, token, baseUrl);
   }

   public getWorksheets(): Promise<Sheet[]> {
      const url = "api/public/worksheets";
      return this.doGet(url).then((list: any) => list.sheets);
   }

   public getWorksheetParameters(id: string): Promise<WorksheetParameter[]> {
      const encodedId = WorksheetClient.encodeId(id);
      const url = `api/public/worksheets/parameters/${encodedId}`;
      return this.doGet(url).then((list: any) => list.parameters);
   }

   public getWorksheetMetadata(id: string): Promise<TableData> {
      const encodedId = WorksheetClient.encodeId(id);
      const url = `api/public/worksheets/metadata/${encodedId}`;
      return this.doGet(url);
   }

   public getOpenedWorksheets(): Promise<Sheet[]> {
      const url = "api/public/worksheets/open";
      return this.doGet(url).then((list: any) => list.sheets);
   }

   public openWorksheet(asset: string, parameters: { [p: string]: any }): Promise<Sheet> {
      const request = { asset, parameters };
      const url = "api/public/worksheets/open";
      return this.doPost(url, request);
   }

   public getOpenedWorksheet(id: string): Promise<Sheet> {
      const url = `api/public/worksheets/open/${id}`;
      return this.doGet(url);
   }

   public closeWorksheet(id: string): Promise<void> {
      const url = `api/public/worksheets/open/${id}`;
      return this.doDelete(url);
   }

   public getWorksheetData(id: string, from: number, rows: number): Promise<TableData> {
      const url = `api/public/worksheets/open/${id}/data`;
      const params = { from: `${from}`, rows: `${rows}` };
      return this.doGet(url, params);
   }
}
