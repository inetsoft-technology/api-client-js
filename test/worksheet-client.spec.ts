/*
 * Copyright 2019 InetSoft Technology
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import { WorksheetClient } from "../src/client/sheet/worksheet/worksheet-client";

describe("WorksheetClient", () => {
   let adapter: any;
   let client: WorksheetClient;

   const sheets = [
      {
         asset: "1^2^__NULL__^Examples/Census Data",
         path: "Examples/Census Data",
         label: "Census Data",
         global: true
      },
      {
         asset: "1^2^__NULL__^Examples/Hurricane",
         path: "Examples/Hurricane",
         label: "Hurricane",
         global: true
      },
      {
         asset: "1^2^__NULL__^Examples/Sales Revenue",
         path: "Examples/Sales Revenue",
         label: "Sales Revenue",
         global: true
      },
      {
         asset: "1^2^__NULL__^Examples/Call Center Monitoring",
         path: "Examples/Call Center Monitoring",
         label: "Call Center Monitoring",
         global: true
      }
   ];
   const openSheets = [
      {
         asset: "1^2^__NULL__^Examples/Call Center Monitoring",
         path: "Examples/Call Center Monitoring",
         label: "Call Center Monitoring",
         global: true,
         identifier: "id1"
      },
      {
         asset: "1^2^__NULL__^Examples/Hurricane",
         path: "Examples/Hurricane",
         label: "Hurricane",
         global: true,
         identifier: "id2"
      }
   ];
   const parameters = [
      {
         name: "startCounter",
         type: "integer"
      }
   ];
   const metadata = {
      rowCount: 2,
      colCount: 24,
      headerRowCount: 1,
      headerColCount: 0,
      trailerRowCount: 0,
      trailerColCount: 0,
      colIdentifiers: ["Annual Trends.Year", "Regions.Code", "Regions.Region", "RawData.State", "RawData.RegionCode", "RawData.DivisionCode", "RawData.Land Area", "RawData.Degree Pct", "RawData.Home Owner Rate", "RawData.Travel Time", "RawData.Housing units", "RawData.Households", "RawData.Poverty Rate", "Divisions.Code", "Divisions.Division", "RawData.Population0", "RawData.Median Income0", "RawData.Property Value0", "Annual Trends.Decr1", "Annual Trends.Incr1", null, null, "RawData.Federal spending", null],
      colTypes: [null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null],
      rows: [{values: ["Year", "Code", "Region", "State", "RegionCode", "DivisionCode", "Land Area", "Degree Pct", "Home Owner Rate", "Travel Time", "Housing units", "Households", "Poverty Rate", "Code_1", "Division", "Population0", "Median Income0", "Property Value0", "Decr1", "Incr1", "Population", "Median Income", "Federal spending", "Property Value"]}, {values: [999.99, 999, "XXXXXX", "XXXXX", 999, 999, 999, 999.99, 999.99, 999.99, 999, 999, 999.99, 999, "XXXXXXXX", 999, 999, 999, "XXXXX", 999.99, 999, 999.99, 999, 999.99]}]
   };
   const data = {
      rowCount: 10,
      colCount: 8,
      headerRowCount: 1,
      headerColCount: 0,
      trailerRowCount: 0,
      trailerColCount: 0,
      colIdentifiers: ["name", "type", "year", "day", "datetime", "latitude", "longitude", "wind"],
      colTypes: [null, null, null, null, null, null, null, null],
      rows: [
         {values: ["name", "type", "year", "day", "datetime", "latitude", "longitude", "wind"]},
         {values: ["ANDREW-1986", "NAMED", 1986, 605, 1.9860605E11, 26.2, -75.8, 30]},
         {values: ["ANDREW-1986", "NAMED", 1986, 605, 1.986060506E11, 27.4, -76.0, 30]},
         {values: ["ANDREW-1986", "NAMED", 1986, 605, 1.986060512E11, 28.4, -76.4, 30]},
         {values: ["ANDREW-1986", "NAMED", 1986, 605, 1.986060518E11, 29.1, -77.0, 30]},
         {values: ["ANDREW-1986", "NAMED", 1986, 606, 1.9860606E11, 29.7, -77.5, 35]},
         {values: ["ANDREW-1986", "NAMED", 1986, 606, 1.986060606E11, 30.2, -77.8, 40]},
         {values: ["ANDREW-1986", "NAMED", 1986, 606, 1.9860606120000003E11, 30.7, -78.0, 45]},
         {values: ["ANDREW-1986", "NAMED", 1986, 606, 1.9860606180000003E11, 31.4, -77.9, 45]},
         {values: ["ANDREW-1986", "NAMED", 1986, 607, 1.9860607E11, 31.9, -77.8, 45]}
      ]
   };

   beforeEach(() => {
      adapter = {
         get: jest.fn(),
         post: jest.fn(),
         put: jest.fn(),
         delete: jest.fn(),
         map: jest.fn(),
         mapText: jest.fn(),
         mapBinary: jest.fn()
      };
      client = new WorksheetClient(adapter, () => Promise.resolve("secret"), "http://localhost:8080/sree/");
   });

   describe("when getWorksheets() is called", () => {
      it("then a list of sheets is returned", () => {
         adapter.get.mockReturnValueOnce(Promise.resolve({data: {sheets}}));
         adapter.map.mockImplementation((promise: Promise<any>) => promise.then(response => response.data));
         return client.getWorksheets().then(result => {
            expect(result).toMatchSnapshot();
            expect(adapter.get.mock.calls.length).toBe(1);
            expect(adapter.get.mock.calls[0][0]).toBe("http://localhost:8080/sree/api/public/worksheets");
         });
      });
   });

   describe("when getWorksheetParameters() is called", () => {
      it("then a list of parameters is returned", () => {
         adapter.get.mockReturnValueOnce(Promise.resolve({data: {parameters}}));
         adapter.map.mockImplementation((promise: Promise<any>) => promise.then(response => response.data));
         return client.getWorksheetParameters("1^2^__NULL__^Examples/Call Center Monitoring").then(result => {
            expect(result).toMatchSnapshot();
            expect(adapter.get.mock.calls.length).toBe(1);
            expect(adapter.get.mock.calls[0][0]).toBe("http://localhost:8080/sree/api/public/worksheets/parameters/MV4yXl9fTlVMTF9fXkV4YW1wbGVzL0NhbGwgQ2VudGVyIE1vbml0b3Jpbmc=");
         });
      });
   });

   describe("when getWorksheetMetadata() is called", () => {
      it("then a table containing the column metadata is returned", () => {
         adapter.get.mockReturnValueOnce(Promise.resolve({data: metadata}));
         adapter.map.mockImplementation((promise: Promise<any>) => promise.then(response => response.data));
         return client.getWorksheetMetadata("1^2^__NULL__^Examples/Census Data").then(result => {
            expect(result).toMatchSnapshot();
            expect(adapter.get.mock.calls.length).toBe(1);
            expect(adapter.get.mock.calls[0][0]).toBe("http://localhost:8080/sree/api/public/worksheets/metadata/MV4yXl9fTlVMTF9fXkV4YW1wbGVzL0NlbnN1cyBEYXRh");
         });
      });
   });

   describe("when getOpenedWorksheets() is called", () => {
      it("then a list of sheets is returned", () => {
         adapter.get.mockReturnValueOnce(Promise.resolve({data: {sheets: openSheets}}));
         adapter.map.mockImplementation((promise: Promise<any>) => promise.then(response => response.data));
         return client.getOpenedWorksheets().then(result => {
            expect(result).toMatchSnapshot();
            expect(adapter.get.mock.calls.length).toBe(1);
            expect(adapter.get.mock.calls[0][0]).toBe("http://localhost:8080/sree/api/public/worksheets/open");
         });
      });
   });

   describe("when openWorksheet() is called", () => {
      it("then the adapter is called with the correct arguments", () => {
         adapter.post.mockReturnValueOnce(Promise.resolve({data: openSheets[0]}));
         adapter.map.mockImplementation((promise: Promise<any>) => promise.then(response => response.data));
         return client.openWorksheet("1^2^__NULL__^Call Center Monitoring", {p1: "v1", p2: "v2"}).then(result => {
            expect(result).toMatchSnapshot();
            expect(adapter.post.mock.calls[0][0]).toBe("http://localhost:8080/sree/api/public/worksheets/open");
            expect(adapter.post.mock.calls[0][1]).toMatchSnapshot();
         });
      });
   });

   describe("when getOpenedWorksheet() is called", () => {
      it("then the opened worksheet is returned", () => {
         adapter.get.mockReturnValueOnce(Promise.resolve({data: openSheets[0]}));
         adapter.map.mockImplementation((promise: Promise<any>) => promise.then(response => response.data));
         return client.getOpenedWorksheet("id1").then(result => {
            expect(result).toMatchSnapshot();
            expect(adapter.get.mock.calls.length).toBe(1);
            expect(adapter.get.mock.calls[0][0]).toBe("http://localhost:8080/sree/api/public/worksheets/open/id1");
         });
      });
   });

   describe("when closeWorksheet() is called", () => {
      it("then the adapter is called with the correct arguments", () => {
         adapter.delete.mockReturnValueOnce(Promise.resolve({}));
         adapter.map.mockImplementation((response: Promise<any>) => response);
         return client.closeWorksheet("id1").then(() => {
            expect(adapter.delete.mock.calls.length).toBe(1);
            expect(adapter.delete.mock.calls[0][0]).toBe("http://localhost:8080/sree/api/public/worksheets/open/id1");
         });
      });
   });

   describe("when getWorksheetData() is called", () => {
      it("then the adapter is called with the correct arguments and the data is returned", () => {
         adapter.get.mockReturnValueOnce(Promise.resolve({data}));
         adapter.map.mockImplementation((promise: Promise<any>) => promise.then(response => response.data));
         return client.getWorksheetData("id2", 0, 10).then(result => {
            expect(result).toMatchSnapshot();
            expect(adapter.get.mock.calls.length).toBe(1);
            expect(adapter.get.mock.calls[0][0]).toBe("http://localhost:8080/sree/api/public/worksheets/open/id2/data");
            expect(adapter.get.mock.calls[0][1]).toMatchSnapshot();
         });
      });
   });
});
