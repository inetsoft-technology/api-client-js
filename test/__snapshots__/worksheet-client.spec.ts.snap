// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WorksheetClient when getOpenedWorksheet() is called then the opened worksheet is returned 1`] = `
Object {
  "asset": "1^2^__NULL__^Examples/Call Center Monitoring",
  "global": true,
  "identifier": "id1",
  "label": "Call Center Monitoring",
  "path": "Examples/Call Center Monitoring",
}
`;

exports[`WorksheetClient when getOpenedWorksheets() is called then a list of sheets is returned 1`] = `
Array [
  Object {
    "asset": "1^2^__NULL__^Examples/Call Center Monitoring",
    "global": true,
    "identifier": "id1",
    "label": "Call Center Monitoring",
    "path": "Examples/Call Center Monitoring",
  },
  Object {
    "asset": "1^2^__NULL__^Examples/Hurricane",
    "global": true,
    "identifier": "id2",
    "label": "Hurricane",
    "path": "Examples/Hurricane",
  },
]
`;

exports[`WorksheetClient when getWorksheetData() is called then the adapter is called with the correct arguments and the data is returned 1`] = `
Object {
  "colCount": 8,
  "colIdentifiers": Array [
    "name",
    "type",
    "year",
    "day",
    "datetime",
    "latitude",
    "longitude",
    "wind",
  ],
  "colTypes": Array [
    null,
    null,
    null,
    null,
    null,
    null,
    null,
    null,
  ],
  "headerColCount": 0,
  "headerRowCount": 1,
  "rowCount": 10,
  "rows": Array [
    Object {
      "values": Array [
        "name",
        "type",
        "year",
        "day",
        "datetime",
        "latitude",
        "longitude",
        "wind",
      ],
    },
    Object {
      "values": Array [
        "ANDREW-1986",
        "NAMED",
        1986,
        605,
        198606050000,
        26.2,
        -75.8,
        30,
      ],
    },
    Object {
      "values": Array [
        "ANDREW-1986",
        "NAMED",
        1986,
        605,
        198606050600,
        27.4,
        -76,
        30,
      ],
    },
    Object {
      "values": Array [
        "ANDREW-1986",
        "NAMED",
        1986,
        605,
        198606051200,
        28.4,
        -76.4,
        30,
      ],
    },
    Object {
      "values": Array [
        "ANDREW-1986",
        "NAMED",
        1986,
        605,
        198606051800,
        29.1,
        -77,
        30,
      ],
    },
    Object {
      "values": Array [
        "ANDREW-1986",
        "NAMED",
        1986,
        606,
        198606060000,
        29.7,
        -77.5,
        35,
      ],
    },
    Object {
      "values": Array [
        "ANDREW-1986",
        "NAMED",
        1986,
        606,
        198606060600,
        30.2,
        -77.8,
        40,
      ],
    },
    Object {
      "values": Array [
        "ANDREW-1986",
        "NAMED",
        1986,
        606,
        198606061200.00003,
        30.7,
        -78,
        45,
      ],
    },
    Object {
      "values": Array [
        "ANDREW-1986",
        "NAMED",
        1986,
        606,
        198606061800.00003,
        31.4,
        -77.9,
        45,
      ],
    },
    Object {
      "values": Array [
        "ANDREW-1986",
        "NAMED",
        1986,
        607,
        198606070000,
        31.9,
        -77.8,
        45,
      ],
    },
  ],
  "trailerColCount": 0,
  "trailerRowCount": 0,
}
`;

exports[`WorksheetClient when getWorksheetData() is called then the adapter is called with the correct arguments and the data is returned 2`] = `
Object {
  "from": "0",
  "rows": "10",
}
`;

exports[`WorksheetClient when getWorksheetMetadata() is called then a table containing the column metadata is returned 1`] = `
Object {
  "colCount": 24,
  "colIdentifiers": Array [
    "Annual Trends.Year",
    "Regions.Code",
    "Regions.Region",
    "RawData.State",
    "RawData.RegionCode",
    "RawData.DivisionCode",
    "RawData.Land Area",
    "RawData.Degree Pct",
    "RawData.Home Owner Rate",
    "RawData.Travel Time",
    "RawData.Housing units",
    "RawData.Households",
    "RawData.Poverty Rate",
    "Divisions.Code",
    "Divisions.Division",
    "RawData.Population0",
    "RawData.Median Income0",
    "RawData.Property Value0",
    "Annual Trends.Decr1",
    "Annual Trends.Incr1",
    null,
    null,
    "RawData.Federal spending",
    null,
  ],
  "colTypes": Array [
    null,
    null,
    null,
    null,
    null,
    null,
    null,
    null,
    null,
    null,
    null,
    null,
    null,
    null,
    null,
    null,
    null,
    null,
    null,
    null,
    null,
    null,
    null,
    null,
  ],
  "headerColCount": 0,
  "headerRowCount": 1,
  "rowCount": 2,
  "rows": Array [
    Object {
      "values": Array [
        "Year",
        "Code",
        "Region",
        "State",
        "RegionCode",
        "DivisionCode",
        "Land Area",
        "Degree Pct",
        "Home Owner Rate",
        "Travel Time",
        "Housing units",
        "Households",
        "Poverty Rate",
        "Code_1",
        "Division",
        "Population0",
        "Median Income0",
        "Property Value0",
        "Decr1",
        "Incr1",
        "Population",
        "Median Income",
        "Federal spending",
        "Property Value",
      ],
    },
    Object {
      "values": Array [
        999.99,
        999,
        "XXXXXX",
        "XXXXX",
        999,
        999,
        999,
        999.99,
        999.99,
        999.99,
        999,
        999,
        999.99,
        999,
        "XXXXXXXX",
        999,
        999,
        999,
        "XXXXX",
        999.99,
        999,
        999.99,
        999,
        999.99,
      ],
    },
  ],
  "trailerColCount": 0,
  "trailerRowCount": 0,
}
`;

exports[`WorksheetClient when getWorksheetParameters() is called then a list of parameters is returned 1`] = `
Array [
  Object {
    "name": "startCounter",
    "type": "integer",
  },
]
`;

exports[`WorksheetClient when getWorksheets() is called then a list of sheets is returned 1`] = `
Array [
  Object {
    "asset": "1^2^__NULL__^Examples/Census Data",
    "global": true,
    "label": "Census Data",
    "path": "Examples/Census Data",
  },
  Object {
    "asset": "1^2^__NULL__^Examples/Hurricane",
    "global": true,
    "label": "Hurricane",
    "path": "Examples/Hurricane",
  },
  Object {
    "asset": "1^2^__NULL__^Examples/Sales Revenue",
    "global": true,
    "label": "Sales Revenue",
    "path": "Examples/Sales Revenue",
  },
  Object {
    "asset": "1^2^__NULL__^Examples/Call Center Monitoring",
    "global": true,
    "label": "Call Center Monitoring",
    "path": "Examples/Call Center Monitoring",
  },
]
`;

exports[`WorksheetClient when openWorksheet() is called then the adapter is called with the correct arguments 1`] = `
Object {
  "asset": "1^2^__NULL__^Examples/Call Center Monitoring",
  "global": true,
  "identifier": "id1",
  "label": "Call Center Monitoring",
  "path": "Examples/Call Center Monitoring",
}
`;

exports[`WorksheetClient when openWorksheet() is called then the adapter is called with the correct arguments 2`] = `
Object {
  "asset": "1^2^__NULL__^Call Center Monitoring",
  "parameters": Object {
    "p1": "v1",
    "p2": "v2",
  },
}
`;
